### ISSUE TITLE
License wrongly detected as GPL 3

### ISSUE THREAD

**Issue by nnobelis** [nexB/scancode-toolkit#2311]
When trying to scan [this file](https://cryptography.gmu.edu:4443/XBX/FreeRTOS/raw/b1796dec343ec1a521f9d1f9a752a2c268451fa5/Demo/CORTEX_LPC1768_GCC_Rowley/LPC1700_Startup.s) with latest scancode, It get the following license :
```
{
          "key": "gpl-3.0-plus",
          "score": 35.0,
          "name": "GNU General Public License 3.0 or later",
          "short_name": "GPL 3.0 or later",
          "category": "Copyleft",
          "is_exception": false,
          "owner": "Free Software Foundation (FSF)",
          "homepage_url": "http://www.gnu.org/licenses/gpl-3.0-standalone.html",
          "text_url": "http://www.gnu.org/licenses/gpl-3.0-standalone.html",
          "reference_url": "https://enterprise.dejacode.com/urn/urn:dje:license:gpl-3.0-plus",
          "spdx_license_key": "GPL-3.0-or-later",
          "spdx_url": "https://spdx.org/licenses/GPL-3.0-or-later",
          "start_line": 4,
          "end_line": 8,
          "matched_rule": {
            "identifier": "gpl-3.0-plus_or_commercial-option_2.RULE",
            "license_expression": "gpl-3.0-plus OR commercial-license",
            "licenses": [
              "gpl-3.0-plus",
              "commercial-license"
            ],
            "is_license_text": false,
            "is_license_notice": true,
            "is_license_reference": false,
            "is_license_tag": false,
            "matcher": "3-seq",
            "rule_length": 80,
            "matched_length": 28,
            "match_coverage": 35.0,
            "rule_relevance": 100
          }
        },
```
### Description

How can ScanCode assume this is GPL3 ? There is no mention of 3 in the lines :

```
 * This file may be distributed under the terms of the License Agreement     *
 * provided with this software.     

```
 
?
Thanks in advance
Nicolas

### How To Reproduce

./scancode -clpeui LPC1700_Startup.s --json-pp output.json

### System configuration

ScanCode 3.2.3 on Linux.

**Comment by pombredanne** [comment]
@nnobelis Thank you for the report!
If I run this scan I can get extra details on the detection:
`scancode -l --license-text --license-text-diagnostics --json-pp output.json LPC1700_Startup.s`
and I get the results below
ScanCode is led to believe that 
```
 * This file may be distributed under the terms of the License Agreement     *
 * provided with this software.                                              *
 *                                                                           *
 * THIS FILE IS PROVIDED AS IS WITH NO WARRANTY OF ANY KIND, INCLUDING THE   *
 * WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE. *
```
is similar enough to https://github.com/nexB/scancode-toolkit/blob/develop/src/licensedcode/data/rules/gpl-3.0-plus_or_commercial-option_2.RULE at https://github.com/nexB/scancode-toolkit/blob/develop/src/licensedcode/data/rules/gpl-3.0-plus_or_commercial-option_2.RULE when this is is rather some generic notice reference to some unknown license.
The (fairly easy) fix will be to add a new rule for that text 

The scans details with --license-text and --license-text-diagnostics are:
```JSON
...

  "files": [
    {
      "path": "LPC1700_Startup.s",
      "type": "file",
      "licenses": [
        {
          "key": "unknown",
          "score": 11.0,
          "name": "Unknown license detected but not recognized",
          "short_name": "unknown",
          "category": "Unstated License",
          "is_exception": false,
          "owner": "Unspecified",
          "homepage_url": null,
          "text_url": "",
          "reference_url": "https://enterprise.dejacode.com/urn/urn:dje:license:unknown",
          "scancode_text_url": "https://github.com/nexB/scancode-toolkit/tree/develop/src/licensedcode/data/licenses/unknown.LICENSE",
          "scancode_data_url": "https://github.com/nexB/scancode-toolkit/tree/develop/src/licensedcode/data/licenses/unknown.yml",
          "spdx_license_key": "LicenseRef-scancode-unknown",
          "spdx_url": "https://github.com/nexB/scancode-toolkit/tree/develop/src/licensedcode/data/licenses/unknown.LICENSE",
          "start_line": 4,
          "end_line": 4,
          "matched_rule": {
            "identifier": "lead-in_unknown_53.RULE",
            "license_expression": "unknown",
            "licenses": [
              "unknown"
            ],
            "is_license_text": false,
            "is_license_notice": false,
            "is_license_reference": true,
            "is_license_tag": false,
            "matcher": "2-aho",
            "rule_length": 2,
            "matched_length": 2,
            "match_coverage": 100.0,
            "rule_relevance": 11
          },
          "matched_text": "distributed under"
        },
        {
          "key": "gpl-3.0-plus",
          "score": 35.0,
          "name": "GNU General Public License 3.0 or later",
          "short_name": "GPL 3.0 or later",
          "category": "Copyleft",
          "is_exception": false,
          "owner": "Free Software Foundation (FSF)",
          "homepage_url": "http://www.gnu.org/licenses/gpl-3.0-standalone.html",
          "text_url": "http://www.gnu.org/licenses/gpl-3.0-standalone.html",
          "reference_url": "https://enterprise.dejacode.com/urn/urn:dje:license:gpl-3.0-plus",
          "scancode_text_url": "https://github.com/nexB/scancode-toolkit/tree/develop/src/licensedcode/data/licenses/gpl-3.0-plus.LICENSE",
          "scancode_data_url": "https://github.com/nexB/scancode-toolkit/tree/develop/src/licensedcode/data/licenses/gpl-3.0-plus.yml",
          "spdx_license_key": "GPL-3.0-or-later",
          "spdx_url": "https://spdx.org/licenses/GPL-3.0-or-later",
          "start_line": 4,
          "end_line": 8,
          "matched_rule": {
            "identifier": "gpl-3.0-plus_or_commercial-option_2.RULE",
            "license_expression": "gpl-3.0-plus OR commercial-license",
            "licenses": [
              "gpl-3.0-plus",
              "commercial-license"
            ],
            "is_license_text": false,
            "is_license_notice": true,
            "is_license_reference": false,
            "is_license_tag": false,
            "matcher": "3-seq",
            "rule_length": 80,
            "matched_length": 28,
            "match_coverage": 35.0,
            "rule_relevance": 100
          },
          "matched_text": "License Agreement     *\n * provided with [this] software.                                              *\n *                                                                           *\n * THIS FILE IS PROVIDED AS IS WITH NO WARRANTY OF ANY KIND, INCLUDING THE   *\n * WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE. *"
        },
        {
          "key": "commercial-license",
          "score": 35.0,
          "name": "Commercial License",
          "short_name": "Commercial License",
          "category": "Commercial",
          "is_exception": false,
          "owner": "Unspecified",
          "homepage_url": null,
          "text_url": "",
          "reference_url": "https://enterprise.dejacode.com/urn/urn:dje:license:commercial-license",
          "scancode_text_url": "https://github.com/nexB/scancode-toolkit/tree/develop/src/licensedcode/data/licenses/commercial-license.LICENSE",
          "scancode_data_url": "https://github.com/nexB/scancode-toolkit/tree/develop/src/licensedcode/data/licenses/commercial-license.yml",
          "spdx_license_key": "LicenseRef-scancode-commercial-license",
          "spdx_url": "https://github.com/nexB/scancode-toolkit/tree/develop/src/licensedcode/data/licenses/commercial-license.LICENSE",
          "start_line": 4,
          "end_line": 8,
          "matched_rule": {
            "identifier": "gpl-3.0-plus_or_commercial-option_2.RULE",
            "license_expression": "gpl-3.0-plus OR commercial-license",
            "licenses": [
              "gpl-3.0-plus",
              "commercial-license"
            ],
            "is_license_text": false,
            "is_license_notice": true,
            "is_license_reference": false,
            "is_license_tag": false,
            "matcher": "3-seq",
            "rule_length": 80,
            "matched_length": 28,
            "match_coverage": 35.0,
            "rule_relevance": 100
          },
          "matched_text": "License Agreement     *\n * provided with [this] software.                                              *\n *                                                                           *\n * THIS FILE IS PROVIDED AS IS WITH NO WARRANTY OF ANY KIND, INCLUDING THE   *\n * WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE. *"
        }
      ],
      "license_expressions": [
        "unknown",
        "gpl-3.0-plus OR commercial-license"
      ],
      "percentage_of_license_text": 2.37,
      "scan_errors": []
    }
  ]
}
```

**Comment by nnobelis** [comment]
Thanks a lot for the thorough explanation.

> The (fairly easy) fix will be to add a new rule for that text

So, just to be clear, you will add the new rule in scancode ?

**Comment by pombredanne** [comment]
@nnobelis yes, I have something in a branch ready. I made a single commit for this as this shows you how you could fix similar things in the future.

**Comment by sschuberth** [comment]
Is this issue already fixed in a release or at least the `develop` branch? If not, I believe we should keep it open until the fix is generally available.

**Comment by nnobelis** [comment]
This is currently fixed in two branches :
`origin/license-updates-2020-10`, `origin/release-20-12-prep`

**Comment by pombredanne** [comment]
Closing as this has been released since v21.2.9
